hyp3_testing/compare.py

```diff
@@ -48,6 +49,46 @@ def values_are_within_tolerance(reference: np.array, secondary: np.array, atol:
         )
 
 
+def _read_as_maskeddata(tif):
+    ds = gdal.Open(tif)
+    band = ds.GetRasterBand(1)
+    mk_band = band.GetMaskBand()
+
+    # True in the mask means the element is valid
+    return ma.array(band.ReadAsArray(), mask=mk_band.ReadAsArray() == 0)
+
+
+def _assert_within_statistic(main_tif, deve_tif, mask_rate, value_range_rate):
+    data_main = _read_as_maskeddata(main_tif)
+    data_deve = _read_as_maskeddata(deve_tif)
+
+    # compare mask
+    valid_mask_and = np.bitwise_and(~data_main.mask, ~data_deve.mask)
+    valid_mask_or = np.bitwise_or(~data_main.mask, ~data_deve.mask)
+    msk_rate = valid_mask_and.sum()/valid_mask_or.sum()
+
+    if msk_rate <= mask_rate:
+        raise AssertionError(
+            f'Two masks match with less than {mask_rate}')
+
+    # compare statistic of data
+    diff = data_main - data_deve
+
+    if diff.max() != diff.min() and 6*diff.std()/(diff.max() - diff.min()) > value_range_rate:
```

Comment:
@cirrusasf I like this approach, but I think we should use the [Kolmogorov–Smirnov test](https://en.wikipedia.org/wiki/Kolmogorov–Smirnov_test) instead of a standard T-test. The K-S test is very similar to a standard T-test, but it does not assume that the data has a normal distribution (and some of our datasets are definitely not normally distributed). Here's a [SciPy function](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ks_2samp.html) we can use to perform this test.

Reply:
great! I will check it out.